app/code/Magento/MediaGallery/Model/Asset/Command/Save.php

```diff
@@ -68,12 +70,39 @@ public function execute(AssetInterface $mediaAsset): int
             $connection = $this->resourceConnection->getConnection();
             $tableName = $this->resourceConnection->getTableName(self::TABLE_MEDIA_GALLERY_ASSET);
 
-            $connection->insertOnDuplicate($tableName, $this->extractor->extract($mediaAsset, AssetInterface::class));
+            $connection->insertOnDuplicate(
+                $tableName,
+                $this->filterData($this->objectProcessor->buildOutputDataArray($mediaAsset, AssetInterface::class))
+            );
             return (int) $connection->lastInsertId($tableName);
         } catch (\Exception $exception) {
             $this->logger->critical($exception);
             $message = __('An error occurred during media asset save: %1', $exception->getMessage());
             throw new CouldNotSaveException($message, $exception);
         }
     }
+
+    /**
+     * Filter data to get flat array without null values
+     *
+     * @param array $data
+     * @return array
+     */
+    private function filterData(array $data): array
+    {
+        $filteredData = [];
+        foreach ($data as $key => $value) {
+            if ($value === null) {
```

Comment:
Merge if conditions

Reply:
Why?

Reply:
Save hardcoded list of fields instead of extracting and filtering data